src/shadowbox/server/shared_metrics.ts

```diff
@@ -82,18 +94,58 @@ export class PrometheusUsageMetrics implements UsageMetrics {
 
   constructor(private prometheusClient: PrometheusClient) {}
 
+  private async queryUsage(metric: string, deltaSecs: number): Promise<PrometheusQueryResult> {
+    const query = `
+      sum(increase(${metric}[${deltaSecs}s]))
```

Comment:
We should escape the metric name here. The deltaSecs is probably ok because it's a number.

From https://prometheus.io/docs/instrumenting/writing_exporters/#naming:

> Only [a-zA-Z0-9:_] are valid in metric names.

Maybe a simple replacement or check is fine.

Reply:
Actually this name is not quite right. It's a time series selector, which can include a label filter. E.g.  `shadowsocks_data_bytes_per_location{dir=~"p>t|p<t"}`. Let me rename it to make that clear.